### PR TITLE
[CheckboxListTile] exposes contentPadding property of ListTile.

### LINKED ISSUE #49688: CheckboxListTile add contentPadding
## Use case

In the same way as in ListTile there is the contentPadding property to adjust the property of the internal padding add it in CheckBoxListTile.

## Proposal

Add the contentPadding property in CheckBoxListTile.

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Exposes `contentPadding` property of `ListTile` so that users can modify the default `contentPadding`.

## Related Issues
Fixes #49688

## Tests

I added the following tests:
- Added test to check contentPadding is properly applied to `CheckboxListTitle`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/checkbox_list_tile.dart [hunk 1]
```diff
@@ -267,6 +267,7 @@ class CheckboxListTile extends StatelessWidget {
     this.selected = false,
     this.controlAffinity = ListTileControlAffinity.platform,
     this.autofocus = false,
+    this.contentPadding
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/checkbox_list_tile.dart:270]: nit: trailing comma
```suggestion
    this.contentPadding,
```

File: packages/flutter/lib/src/material/checkbox_list_tile.dart [hunk 2]
```diff
@@ -356,6 +357,11 @@ class CheckboxListTile extends StatelessWidget {
   /// {@macro flutter.widgets.Focus.autofocus}
   final bool autofocus;
 
+  /// Defines insets for [Checkbox], [title], [subtitle], and [secondary].
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/checkbox_list_tile.dart:360]: Just a suggestion for improvement to the wording, but you can do something else that might make more sense to you. I'm trying to use wording to distinguish between the insets for the entire tile vs. padding that insets every single widget in the tile (ie. 10px padding surrounding the tile's entire contents vs 10px padding surrounding checkbox, 10px padding surrounding title, etc).
```suggestion
  /// Defines insets surrounding the tile's contents. 
  ///
  /// This value will surround the [Checkbox], [title], [subtitle], and [secondary]
  /// widgets in [CheckboxListTile].
```

**Review by shihaohong — APPROVED** [review]
LGTM with some nits

**Comment by AyushBherwani1998** [comment]
> 
> 
> LGTM with some nits

Thank you for the review 😄 . I have committed the suggestions.

**Pull request merged**